core/gather/driver/target-manager.js

```diff
@@ -97,10 +103,21 @@ class TargetManager extends ProtocolEventEmitter {
     return this._findSession(rootSessionId);
   }
 
+  executionContexts() {
```

Comment:
Explicit `mainFrameExecutionContexts` is a good call, but FWIW I was imagining a completely different interface for general access to all execution contexts (maybe find the target of interest first, then get its execution contexts), so the architecture space seems completely open. Could we drop this for now and then open up discussion when we need it?

Reply:
Sure.